### PR TITLE
SAK-32215 Prevent deadlock at startup due to quartz misfire handler

### PR DESCRIPTION
**Pull request by buckett**
If quartz tried to run some misfired jobs while the component manager was still starting up it would deadlock the startup of Sakai.

This introduces the Spring lifecycle to our application context and uses it to start the quartz scheduler once the application context has been refreshed.

### CODE REVIEW

**Comment by ottenhoff** [comment]
Ok seems fine. @ern how about a quick review?

File: jobscheduler/scheduler-component-shared/src/java/org/sakaiproject/component/app/scheduler/SchedulerManagerImpl.java [hunk 1]
```diff
@@ -745,4 +735,39 @@ public boolean isStartScheduler() {
...
+                LOG.error("Failed to start the scheduler.", e);
+            }
+        } else {
+            LOG.info("Scheduler Not Started, startScheduler=false");
+        }
+    }
+
+    @Override
+    public void stop() {
+        try{
```
Comment (ern) [inline jobscheduler/scheduler-component-shared/src/java/org/sakaiproject/component/app/scheduler/SchedulerManagerImpl.java]: Probably should make a call to super() as your overriding the application context stop().

File: jobscheduler/scheduler-component-shared/src/java/org/sakaiproject/component/app/scheduler/SchedulerManagerImpl.java [hunk 2]
```diff
@@ -745,4 +735,39 @@ public boolean isStartScheduler() {
    public void setStartScheduler(boolean startScheduler) {
        this.startScheduler = startScheduler;
    }
+
+    @Override
+    public void start() {
+        if (isStartScheduler()) {
```
Comment (ern) [inline jobscheduler/scheduler-component-shared/src/java/org/sakaiproject/component/app/scheduler/SchedulerManagerImpl.java]: Probably should make a call to super() as your overriding the application context start()

**Comment by ern** [comment]
LGTM, good use on using the ApplicationContext lifecycle 👍

**Pull request merged**